src/etc/tidy.py

```diff
@@ -146,6 +148,9 @@ def interesting_file(f):
                     if not hsh.startswith(rev):
                         report_err("snapshot out of date (" + date
                             + "): " + line)
+                match = re.match(r'^.*#!\[unstable.*\]', line)
+                if match:
+                    needs_unstable_attr.discard(filename)
                 else:
```

Comment:
It seems that this changes when the below check is executed. I know basically nothing about tidy but it seems that the way this is written will hide errors for erroneous `SNAP` comments (unless they happen to be on an `#![unstable...]` line).